mycore-ocfl/src/main/java/org/mycore/ocfl/metadata/MCROCFLXMLMetadataManager.java

```diff
@@ -151,6 +162,39 @@ public void delete(MCRObjectID mcrid, Date date, String user) throws MCRPersiste
         });
     }
 
+    public void purge(MCRObjectID mcrid, Date date, String user, boolean skipPermCheck) {
+        String ocflObjectID = getOCFLObjectID(mcrid);
+        if (!exists(mcrid)) {
+            throw new MCRUsageException("Cannot delete nonexistent object '" + ocflObjectID + "'");
+        }
+
+        if (!skipPermCheck && !MCRAccessManager.checkPermission(mcrid, MCRAccessManager.PERMISSION_HISTORY_DELETE)) {
+            throw new MCRException("You are not authorized to delete '" + ocflObjectID + "' and its history!");
+        }
+
+        OcflRepository repo = getRepository();
+        repo.purgeObject(ocflObjectID);
+        // VersionInfo headVersion = repo.describeObject(ocflObjectID).getHeadVersion().getVersionInfo();
+        // TODO find a good way to do this
+        // repo.updateObject(ObjectVersionId.head(ocflObjectID), buildVersionInfo("Purged", date, user), null);
+    }
+
+    public void restore(MCRObjectID mcrid, String revision) {
+        String prefix = "derivate".equals(mcrid.getTypeId()) ? MCROCFLObjectIDPrefixHelper.MCRDERIVATE
```

Comment:
Also regarding "The non-constant portion of an equals check generally comes first.", this is not true.
Having a non-constant first comes with the risk of having a `NullPointerException` when trying to run `.equals()` on `null`, while `string.equals(null)` will just return `false`, since they don't equal.

Plus see comment below, we don't compare two strings with `Objects.equals()`, `Object.equals()` compares memory references while `String.equals()` compares the content of a string.

Reply:
@sonatype-lift ignore